reaper.go

```diff
@@ -21,7 +22,8 @@ type reaper struct {
 	sync.RWMutex
 
 	chansLock     sync.RWMutex
-	exitCodeChans map[int]chan int
+	exitCodeChans map[string]chan int
+	pidToExecID   map[int]string
```

Comment:
I would like to keep using `exitCodeChans map[int]chan int` with an adapted reaper:

1) the reaper loop automatically delete the entry when the pid is reaped.
  (simplify codes and also fix the problem of reusing pid by the kernel).
2) the `chan int` is also saved on the process struct for WaitProcess(). 
   (don't need to search it from the reaper again)
3) reaper.start() returns `chan int`, the caller should wait for it before exec.Cmd.Wait() or other Wait().
4) remove reaper.wait() and corresponding interfaces.

Reply:
@laijs thanks for your comments, I agree it makes things simpler. I have addressed 1, 2, 3, but not 4 because I have made the choice to pass the `exitCodeCh` as an argument of the reaper function `wait()`, so that we can keep the generic call here. No need to duplicate from the caller.